workshops/fabric-e2e-rag/workshop.md

```diff
@@ -0,0 +1,480 @@
+2. Embedding Generation and Storage: Learn how to generate embeddings for the chunks using SynapseML and [Azure OpenAI Services](https://azure.microsoft.com/products/cognitive-services/openai-service), store the embeddings in a vector store using [Azure Cognitive Search](https://azure.microsoft.com/products/search), and search the vector store to answer the user’s question.
+3. Question Answering Pipeline: Learn how to retrieve relevant document based on the user’s question and provide the answer using [Langchain](https://python.langchain.com/en/latest/index.html#).
+
+To get an indepth understanding of the RAG framework, refer to [this workshop](https://moaw.dev/workshop/gh:azure-samples/azure-openai-rag-workshop/base/docs/)
+
+## Prerequisites
+
+To follow along with this work begin by Right-click or `Ctrl + Click` on the button below to open the Azure Portal in a new tab and begin deployment of the necessary resources.
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fmicrosoft%2moaw%2Fmain.json)
+
+This deploys the following resources:
+1. Azure OpenAI Service
+2. Azure AI Search
+3. Azure AI Services
+4. Azure Key Vault
+
+<div class="info" data-title="Note">
+  
+  > The Azure Open AI deployment also includes the deployment of the ***gpt-4*** and ***text-embedding-ada-002 models*** 
+</div>
+
+---
+
+# Loading and Preprocessing PDF Documents
+In this section, we'll load the PDF document into a spark dataframe then read the documents using the Azure AI Document Intelligence in Azure AI Services. 
+
+After extracting the text from the PDF documents, we'll use SynapseML to split the documents into chunks. The document is split into chunks to allow for more granular representation and processing of the document content.
+
+To begin with [create a Lakehouse](https://learn.microsoft.com/en-us/fabric/data-engineering/tutorial-build-lakehouse#create-a-lakehouse?WT.mc_id=data-114676-jndemenge) in your Microsoft Fabric workspace. Next in the Data Engineering workload create a new Notebook to continue.
+
+We'll then need to install the necessary libraries. We'll use the [semantic-kernel](https://learn.microsoft.com/en-us/semantic-kernel/overview?WT.mc_id=data-114676-jndemenge) library to build the RAG application with models from Azure OpenAI. 
+
+Copy the following code into a new cell and run it to install the libraries.
+
+```python
+%pip install --quiet semantic-kernel
+```
+Next we'll need to provide the keys for Azure AI Services and Azure OpenAI to access the respective services.Copy the values from the Azure Portal and paste them into the following code cell.
+
```

Comment:
I would at least put a comment to explain that - in a production scenario - we should store the credentials in Key Vault and use `mssparkutils`